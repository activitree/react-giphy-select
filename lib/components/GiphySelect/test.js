"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _index = _interopRequireDefault(require("./index"));

describe('<GiphySelect />', function () {
  fetch.mockResponse(JSON.stringify({
    data: [],
    pagination: {
      count: 0,
      offset: 0,
      total: 0
    }
  }));
  test('render component', function () {
    var spy = jest.spyOn(_index.default.prototype, 'render');
    var wrapper = shallow( /*#__PURE__*/React.createElement(_index.default, null));
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });
  test('snapshot match', function () {
    var wrapper = shallow( /*#__PURE__*/React.createElement(_index.default, null));
    expect(wrapper).toMatchSnapshot();
  });
  test('set custom theme', function () {
    var customTheme = {
      select: 'customSelect',
      selectInput: 'customSelectInput',
      attribution: 'customAttribution'
    };
    var wrapper = shallow( /*#__PURE__*/React.createElement(_index.default, {
      theme: customTheme
    }));
    expect(Object.keys(customTheme).every(function (key) {
      return wrapper.find(".".concat(customTheme[key])).length;
    })).toBe(true);
  });
  test('set placeholder to search field', function () {
    var customPlaceholder = 'test';
    var wrapper = shallow( /*#__PURE__*/React.createElement(_index.default, {
      placeholder: customPlaceholder
    }));
    expect(wrapper.find('input').prop('placeholder')).toBe(customPlaceholder);
  });
});
//# sourceMappingURL=test.js.map