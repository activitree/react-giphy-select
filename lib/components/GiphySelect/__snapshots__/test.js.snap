// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiphySelect /> snapshot match 1`] = `
<div
  className="react-giphy-select__src-components-GiphySelect-styles__select___25GhN"
  onWheel={[Function]}
>
  <input
    className="react-giphy-select__src-components-GiphySelect-styles__selectInput___18Wcj"
    onChange={[Function]}
    placeholder="Search GIFs"
  />
  <GiphyList
    items={Array []}
    loadNextPage={[Function]}
    onEntrySelect={[Function]}
    renderEntry={[Function]}
    theme={
      Object {
        "attribution": "react-giphy-select__src-components-GiphySelect-styles__attribution___1RUSw",
        "select": "react-giphy-select__src-components-GiphySelect-styles__select___25GhN",
        "selectInput": "react-giphy-select__src-components-GiphySelect-styles__selectInput___18Wcj",
      }
    }
  />
  <div
    className="react-giphy-select__src-components-GiphySelect-styles__attribution___1RUSw"
  >
    Powered by Giphy
  </div>
</div>
`;
